pkg/action/drutils.go

```diff
@@ -0,0 +1,164 @@
+			logEvents(msg, "out")
+			action.Status.Status = storkv1.ActionStatusSuccessful
+		} else {
+			msg := fmt.Sprintf("Failing %s operation as the last mile migration %s failed: status %s", action.Spec.ActionType, migrationObject.Name, migrationObject.Status.Status)
+			log.ActionLog(action).Errorf(msg)
+			action.Status.Status = storkv1.ActionStatusFailed
+			action.Status.Reason = msg
+			ac.recorder.Event(action,
+				v1.EventTypeWarning,
+				string(storkv1.ActionStatusFailed),
+				msg)
+		}
+		ac.updateAction(action)
+		return
+	}
+	log.ActionLog(action).Infof("Last mile migration %s is still running, currently in stage: %s", migrationObject.Name, migrationObject.Status.Stage)
+}
+
+// getDRMode determine the mode of DR either from clusterPair or from the clusterDomains
+func (ac *ActionController) getDRMode(clusterPairName string, namespace string) (DRKind, error) {
+	clusterPair, cpErr := storkops.Instance().GetClusterPair(clusterPairName, namespace)
+	if cpErr == nil {
+		if len(clusterPair.Spec.Options) != 0 {
+			return asyncDR, nil
+		}
+		return syncDR, nil
+	}
+	// since clusterPair get failed, we will fall back to seeing if clusterDomains are present or not
+	currentClusterDomains, cdErr := ac.volDriver.GetClusterDomains()
+	if cdErr != nil {
+		return invalid, fmt.Errorf("unable to determine if the DR plan's mode is async-dr or sync-dr: %v ; %v", cpErr, cdErr)
+	}
+	if len(currentClusterDomains.ClusterDomainInfos) != 0 {
+		return syncDR, nil
+	}
+	return asyncDR, nil
+}
+
+// remoteClusterDomainUpdate updates the remote cluster domain.
+// If activate is true, it activates the domain. If activate is false, it deactivates the domain.
+// The method returns an error if it fails to get cluster domain info or if activate/deactivate call fails.
+func (ac *ActionController) remoteClusterDomainUpdate(activate bool, action *storkv1.Action) error {
+	currentClusterDomains, err := ac.volDriver.GetClusterDomains()
+	if err != nil {
+		return fmt.Errorf("failed to get cluster domain info: %v", err)
+	}
+	log.ActionLog(action).Infof("Count of cluster domains: %v ; Current cluster domains: %v", len(currentClusterDomains.ClusterDomainInfos), currentClusterDomains)
+
+	var remoteDomainName string
+	for _, apiDomainInfo := range currentClusterDomains.ClusterDomainInfos {
+		if apiDomainInfo.Name == currentClusterDomains.LocalDomain {
+			continue
+		}
+		remoteDomainName = apiDomainInfo.Name
+		break
+	}
+
+	if remoteDomainName != "" {
+		cduName := uuid.New()
+		clusterDomainUpdate := &storkv1.ClusterDomainUpdate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cduName,
+			},
+			Spec: storkv1.ClusterDomainUpdateSpec{
+				ClusterDomain: remoteDomainName,
+				Active:        activate,
+			},
+		}
+		if activate {
+			err := ac.volDriver.ActivateClusterDomain(clusterDomainUpdate)
+			if err != nil {
+				return fmt.Errorf("activation of remote cluster domain: %v failed : %v", remoteDomainName, err)
```

Comment:
Let's remove "remote" from here since we are always activating/deactivating the source.

Reply:
okay, will address